### PR TITLE
feat: ARI "ChannelToneDetected" event

### PR DESCRIPTION
**Pull request by gibbz00**
Fixes: #811

Refrained from adding any JSON blob to the message. Perhaps it should be part of a separate PR if anyone wants to build on this later on.

### CODE REVIEW

**Comment by gibbz00** [comment]
cherry-pick-to: none

File: include/asterisk/stasis_channels.h [hunk 1]
```diff
@@ -656,6 +656,14 @@ struct stasis_message_type *ast_channel_talking_start(void);
  */
 struct stasis_message_type *ast_channel_talking_stop(void);
 
+/*!
+ * \since 21.3.2
```
Comment (jcolp) [inline include/asterisk/stasis_channels.h:660]: This is no longer correct, and this change is not commented to be cherry picked against 21 anyway

**Review by jcolp — CHANGES_REQUESTED** [review]
This new JSON event needs to be documented in rest-api/api-docs/events.json and "make ari-stubs" re-run.

The commit message also needs to e expanded and a UserNote added about this addition.

File: include/asterisk/stasis_channels.h (hunk 1 above)
Comment (gibbz00) [reply include/asterisk/stasis_channels.h]: `\since 22` now, is that ok?

**Comment by gibbz00** [comment]
> This new JSON event needs to be documented in rest-api/api-docs/events.json and "make ari-stubs" re-run.

Done, thanks.

> The commit message also needs to e expanded and a UserNote added about this addition.

Updated the commit message, not 100% how much you want to be included.

**Review by jcolp — CHANGES_REQUESTED** [review]
The commit message is rather technical and internal. The following is high level enough and easy for users to understand in the case of UserNote:

feat: ARI "ChannelToneDetected" event

A stasis event is now produced when using the TONE_DETECT dialplan
function. This event is published over ARI using the ChannelToneDetected
event. This change does not make it available over AMI.

Fixes: asterisk#811

UserNote: Setting the TONE_DETECT dialplan function on a channel
in ARI will now cause a ChannelToneDetected ARI event to be raised
when the specified tone is detected.

**Comment by gibbz00** [comment]
Updated, thank you

File: res/res_tonedetect.c [hunk 2]
```diff
@@ -394,6 +396,16 @@ static int detect_callback(struct ast_audiohook *audiohook, struct ast_channel *
 			}
 			ast_debug(1, "TONE_DETECT just got a hit (#%d in this direction, waiting for %d total)\n", now, di->hitsrequired);
 			if (now >= di->hitsrequired) {
+				message = ast_channel_blob_create_from_cache(ast_channel_uniqueid(chan), ast_channel_tone_detect(), NULL);
+
+				if (!message) {
+					ast_log(LOG_ERROR, "unable to create ast_channel_tone_detect message");
```
Comment (jcolp) [inline res/res_tonedetect.c:402]: This should include the channel name.

"Unable to publish tone detected event for ARI on channel '%s'"

**Review by jcolp — CHANGES_REQUESTED** [review]

File: res/res_tonedetect.c (hunk 2 above)
Comment (gibbz00) [reply res/res_tonedetect.c]: Yeah, fixed :)

**Review by jcolp — APPROVED** [review]

**Comment by gtjoseph** [comment]
testsuite-test-pr: 54

**Pull request merged**